### PR TITLE
Clean-up the code base and fix core tests

### PR DESCRIPTION
**Pull request by IwoHerka**
This pull contains:
1. Fix for broken core tests
2. General cleanup of the code base
3. Some added comments

### CODE REVIEW

File: src/name_service/delivery/server/delete.clj [hunk 1]
```diff
@@ -18,7 +18,7 @@
         (state/delete-key-binding key value)
         (utils/action->response))))
 
-; if json-params contain delete, given key is going to be deleted
+; Jf json-params contain delete, given key is going to be deleted
```
Comment (ehofman) [inline src/name_service/delivery/server/delete.clj]: typo

File: src/name_service/delivery/storage/mongo.md [hunk 2]
```diff
@@ -1,14 +1,18 @@
 # mongoDB
 
-Microservice is using [Clojure Monger](http://clojuremongodb.info/) to connect with MongoDB.
+Microservice is using [Clojure Monger](http://clojuremongodb.info/) to
```
Comment (ehofman) [inline src/name_service/delivery/storage/mongo.md]: The mongo.md file should be in the docs module and docs/README.md should navigate to this file

**Review by ehofman — CHANGES_REQUESTED** [review]

File: src/name_service/core/spec.clj [hunk 3]
```diff
@@ -24,6 +24,6 @@
         :args (s/cat :storage ::storage :key ::key :val ::val)
         :ret  boolean?)
 
-(s/fdef entity/fetch
-        :args (s/cat :storage ::storage :key ::key :val ::val)
-        :ret ::storage-result)
\ No newline at end of file
+(s/fdef entity/storage-error?
+        :args (s/cat :result ::storage-result)
+        :ret  boolean?)
```
Comment (pankominek) [inline src/name_service/core/spec.clj]: Add a blank line to the end of the file.

File: src/name_service/core/use_case.clj [hunk 4]
```diff
@@ -1,69 +1,80 @@
...
+        (a/make-error ::e/get keymap)
+        (result->action ::e/get (get keymap key-a))))))
+
+(defn- result->action
+  "Convert result of a storage action to action object."
+  [type result]
+  (let [payload {:result result}]
+    (if (e/storage-error? result)
+      (a/make-error type :storage payload)
+      (a/make-action type payload))))
```
Comment (pankominek) [inline src/name_service/core/use_case.clj]: Add a blank line at the end of the file.

File: src/name_service/delivery/server/append.clj [hunk 5]
```diff
@@ -20,6 +20,5 @@
 
 (defn dispatch [{:keys [json-params]}]
   (if (s/valid? ::spec/post-request json-params)
-    (append json-params)
-    (http/bad-request
-      (s/explain-data ::spec/post-request json-params))))
+    (append-key json-params)
+    (http/bad-request (s/explain-data ::spec/post-request json-params))))
```
Comment (pankominek) [inline src/name_service/delivery/server/append.clj]: Add a blank line at the end of the file.

File: src/name_service/delivery/server/get.clj [hunk 6]
```diff
@@ -15,6 +15,6 @@
 
 (defn dispatch [request]
   (if (s/valid? ::spec/get-request request)
-    (querying request)
+    (get-key request)
     (http/bad-request
-      (s/explain-data ::spec/get-request request))))
+      (s/explain-data ::spec/get-request request))))
```
Comment (pankominek) [inline src/name_service/delivery/server/get.clj]: Add a blank line at the end of the file.

File: src/name_service/delivery/server/http.clj [hunk 7]
```diff
@@ -0,0 +1,10 @@
+(ns name-service.delivery.server.http)
+
+(defn- response
+  [status body & {:as headers}]
+  {:status status
+   :body body
+   :headers (merge {"Content-Type" "application/json"} headers)})
+
+(def ok          (partial response 200))
+(def bad-request (partial response 400))
```
Comment (pankominek) [inline src/name_service/delivery/server/http.clj:10]: Add a blank line at the end of the file.

File: src/name_service/delivery/storage/spec.clj [hunk 8]
```diff
@@ -0,0 +1,6 @@
+(ns name-service.delivery.storage.spec
+  (:require [clojure.spec.alpha :as s]
+            [name-service.delivery.storage.mongo :as mongo]
+            [name-service.core.spec :as cs]))
+
+(s/fdef mongo/make-storage :ret ::cs/storage)
```
Comment (pankominek) [inline src/name_service/delivery/storage/spec.clj]: Add a blank line at the end of the file.

File: src/name_service/delivery/server/utils.clj [hunk 9]
```diff
@@ -27,17 +22,20 @@
...
     (= error :storage)
-    (http/bad-request (json/encode {:error  "Saving data failed."
-                                    :result payload}))
+    (http/bad-request (json/encode {:error "Saving data failed." :result payload}))
+
     (map? error)
     (http/bad-request (json/encode {:error error}))
+
     (nil? error)
     (http/ok (merge-with-payload->json {:error nil} payload))))
```
Comment (pankominek) [inline src/name_service/delivery/server/utils.clj:39]:
```suggestion
    (http/ok (json/encode payload))))
```

Will not display "nil" error with 200 response code.

File: src/name_service/delivery/server/utils.clj [hunk 10]
```diff
@@ -1,20 +1,15 @@
...
+  (-> map keys first name))
 
 (defn first-value-to-str [map]
-  (-> map
-    vals
-    first))
+  (-> map vals first))
 
 (defn merge-with-payload->json [coll payload]
   (json/encode (merge payload coll)))
```
Comment (pankominek) [inline src/name_service/delivery/server/utils.clj:14]: Remove these lines if you don't want to display the "nil" error.

File: src/name_service/delivery/server/append.clj [hunk 11]
```diff
@@ -1,11 +1,11 @@
-(ns name-service.delivery.append
+(ns name-service.delivery.server.append
   (:require [clojure.spec.alpha :as s]
-            [name-service.delivery.http :as http]
-            [name-service.delivery.spec :as spec]
             [name-service.delivery.state :as state]
-            [name-service.delivery.utils :as utils]))
+            [name-service.delivery.server.spec :as spec]
```
Comment (dbisiako) [inline src/name_service/delivery/server/append.clj]: Arrange requirements in  alphabetically.

File: src/name_service/delivery/server/service.clj [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+   ; This can also be your own chain provider/server-fn -- http://pedestal.io/reference/architecture-overview#_chain_provider
+   ::http/type :jetty
+   ::http/host "0.0.0.0"
+   ::http/port 8080
+   ; Options to pass to the container (Jetty)
+   ::http/container-options
+   {:h2c? true
+    :h2? false
+    ;:keystore "test/hp/keystore.jks"
+    ;:key-password "password"
```
Comment (dbisiako) [inline src/name_service/delivery/server/service.clj:54]: I think comments should have space between `;` and sentence.

File: src/name_service/core/spec.clj (hunk 3 above)
Comment (IwoHerka) [reply src/name_service/core/spec.clj]: Why do you want blanks at the end of files?

File: src/name_service/delivery/server/service.clj (hunk 12 above)
Comment (IwoHerka) [reply src/name_service/delivery/server/service.clj:54]: They should, but this is not a comment, it's a commented config.

**Review by mateuszluczak1996 — APPROVED** [review]

File: src/name_service/core/spec.clj (hunk 3 above)
Comment (pankominek) [reply src/name_service/core/spec.clj]: Sorry, I meant a empty line.

**Review by ehofman — APPROVED** [review]

**Review by pankominek — COMMENTED** [review]
Add a empty line at the end of the file: src / name_service / delivery / server / http.clj .

**Review by dbisiako — APPROVED** [review]

**Review by mateuszluczak1996 — APPROVED** [review]

**Pull request merged**